stl/inc/__msvc_format_ucd_tables.hpp

```diff
@@ -86,11 +86,17 @@ struct _Unicode_property_data {
         --_Upper_idx;
         const uint32_t _Lower_bound = _Lower_bounds[_Upper_idx];
         const uint16_t _Data        = _Props_and_size[_Upper_idx];
-        const uint16_t _Size        = static_cast<uint16_t>(_Data & 0x0FFF);
-        const _ValueEnum _Prop      = static_cast<_ValueEnum>((_Data & 0xF000) >> 12);
-        _STL_INTERNAL_CHECK(_Code_point >= _Lower_bound);
-        if (_Code_point < _Lower_bound + _Size) {
-            return _Prop;
+        if constexpr (_Is_binary_property) {
+            if (_Code_point < _Lower_bound + _Data) {
+                return static_cast<_ValueEnum>(0);
+            }
+        } else {
+            const uint16_t _Size   = static_cast<uint16_t>(_Data & 0x0FFF);
+            const _ValueEnum _Prop = static_cast<_ValueEnum>((_Data & 0xF000) >> 12);
+            _STL_INTERNAL_CHECK(_Code_point >= _Lower_bound);
+            if (_Code_point < _Lower_bound + _Size) {
+                return _Prop;

```

Comment:
I'm OK with this, but how much space do we actually save by using these extra 4 bits?

Reply:
The following ranges will need to be split if the range size is capped at 4096:

lower|upper|size
-|-|-
U+3220|U+A48C|29293
U+AC00|U+D7A3|11172
U+17000|U+187F7|6136
U+20000|U+2A6DF|42720
U+2A700|U+2B739|4154
U+2B820|U+2CEA1|5762
U+2CEB0|U+2EBE0|7473
U+30000|U+3134A|4939
U+31350|U+323AF|4192

The split will result in 25 more ranges, which means 25 * 6 = 150 more bytes (if my math is correct).